src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -297,36 +297,42 @@ def _save_export(self, export):
     def create_export(self, export_type, fs_name, pseudo_path, read_only, path, cluster_id):
         if export_type != 'cephfs':
             return -errno.EINVAL,"", f"Invalid export type: {export_type}"
+
+        if not self.check_fs(fs_name):
+            return -errno.EINVAL,"", "Invalid CephFS name"
+
         #TODO Check if valid cluster
         if cluster_id not in self.exports:
             self.exports[cluster_id] = []
 
         self.rados_namespace = cluster_id
-        if not self.check_fs(fs_name) or self._fetch_export(pseudo_path):
-            return -errno.EINVAL,"", "Invalid CephFS name or export already exists"
-
-        ex_id = self._gen_export_id()
-        user_id = f"{cluster_id}{ex_id}"
-        user_out, key = self._create_user_key(user_id, path, fs_name)
-        if isinstance(user_out, int):
-            return user_out, "", key
-
         access_type = "RW"
         if read_only:
             access_type = "R"
 
-        ex_dict = {
-            'path': self.format_path(path),
-            'pseudo': self.format_path(pseudo_path),
-            'cluster_id': cluster_id,
-            'access_type': access_type,
-            'fsal': {"name": "CEPH", "user_id": user_id, "fs_name": fs_name, "sec_label_xattr": ""},
-            'clients': []
-            }
-
-        export = Export.from_dict(ex_id, ex_dict)
-        export.fsal.cephx_key = key
-        self._save_export(export)
+        if not self._fetch_export(pseudo_path):
+            ex_id = self._gen_export_id()
+            user_id = f"{cluster_id}{ex_id}"
+            user_out, key = self._create_user_key(user_id, path, fs_name)
+
+            if isinstance(user_out, int):
+                return user_out, "", key
+
+            ex_dict = {
+                    'path': self.format_path(path),
+                    'pseudo': self.format_path(pseudo_path),
+                    'cluster_id': cluster_id,
+                    'access_type': access_type,
+                    'fsal': {"name": "CEPH", "user_id": user_id,
+                             "fs_name": fs_name, "sec_label_xattr": ""},
+                    'clients': []
+                    }
+
+            export = Export.from_dict(ex_id, ex_dict)
+            export.fsal.cephx_key = key
+            self._save_export(export)
+        else:
+            log.error("Export already exists")
```

Comment:
@varshar16  please also remember that all nfs commands must be idempotent.

Reply:
``` 
build]$ ceph nfs export create cephfs a test /cephfs --readonly true
{
    "bind": "/cephfs",
    "fs": "a",
    "path": "true",
    "cluster": "test",
    "mode": "R"
}
```

Then when I create the same export (i.e., same cluster ID and binding) but with different access mode('RW' instead of 'R'), stdout shows that export with updated access mode has been created. I expected the export dict not to be displayed, and see a message that export already exists.
```
$ ceph nfs export create cephfs a test /cephfs
{
    "bind": "/cephfs",
    "fs": "a",
    "path": "/",
    "cluster": "test",
    "mode": "RW"
}
```
However, the actual export object is not updated with the new access mode,
```
$ rados -p nfs-ganesha -N test get export-1 -
EXPORT {
    export_id = 1;
    path = "true";
    pseudo = "/cephfs";
    access_type = "R";
    squash = "no_root_squash";
    security_label = true;
    protocols = 4;
    transports = "TCP";
    FSAL {
        name = "CEPH";
        user_id = "test1";
        filesystem = "a";
        secret_access_key = "AQB+pcteFu81OhAAXUA0Wnl9taN59LFNC/R7qg==";
    }

}
```